src/components/settings/LanguageSelector.tsx

```diff
@@ -0,0 +1,43 @@
+import { Globe } from 'lucide-react'
+import { useTranslation } from 'react-i18next'
+import { Select, SelectContent, SelectItem, SelectTrigger, SelectValue } from '@/components/ui/select'
+import languages from '@/i18n/languages'
+
+export const LanguageSelector = () => {
+  const { i18n, t } = useTranslation()
+
+  const handleLanguageChange = (languageKey: string) => {
+    i18n.changeLanguage(languageKey)
+  }
+
+  const getCurrentLanguageDescription = () => {
+    const currentLanguage = languages.find((lang) => lang.key === i18n.language)
+    return currentLanguage?.description || 'English'
+  }
+
+  return (
```

Comment:
If a user never selected a wallet, it will be empty.
Should default to English.

![image](https://github.com/user-attachments/assets/637d0508-9eaa-41cc-9759-62c0855e066f)